### PR TITLE
added curPage in datagrid

### LINKED ISSUE #559: DataGrid has no curPage attribute
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
added curPage in dataGrid

## Issue ticket number and link

Closes #559 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

**Review by evshi — CHANGES_REQUESTED** [review]
I'm not sure when you added `skipFirstRender` but why is that there?

**Comment by sagardspeed2** [comment]
> I'm not sure when you added `skipFirstRender` but why is that there?

`skipFirstRender` is needed to just void unnecessary API call on the dataGrid widget register, API should only render on the page change or pageSize change

**Comment by evshi** [comment]
Can you please remove? You don't need it.

File: packages/runtime/src/widgets/DataGrid/DataGrid.tsx [hunk 1]
```diff
@@ -176,7 +180,7 @@ export const DataGrid: React.FC<GridProps> = (props) => {
       setSelectionType,
       setAllowSelection,
       setPageSize,
-      setCurPage,
+      setPage,
```
Comment (evshi) [inline packages/runtime/src/widgets/DataGrid/DataGrid.tsx:183]: This is an API breaking change by renaming the function. You need to keep the key the same.

File: packages/runtime/src/widgets/DataGrid/DataGrid.tsx [hunk 2]
```diff
@@ -164,6 +163,11 @@ export const DataGrid: React.FC<GridProps> = (props) => {
     },
   };
 
+  const setPage = (page: number) => {
```
Comment (evshi) [inline packages/runtime/src/widgets/DataGrid/DataGrid.tsx:166]: Use `useCallback`

File: packages/runtime/src/widgets/DataGrid/DataGrid.tsx [hunk 3]
```diff
@@ -282,6 +246,55 @@ export const DataGrid: React.FC<GridProps> = (props) => {
...
+    (n: number) => {
+      setCurPage(n);
+      onPageChangeActionCallback(n, pageSize || 10);
+    },
+    [pageSize, onPageChangeActionCallback],
+  );
+
+  // update pageSize callback
+  const updatePageSize = useCallback(
+    (n: number) => {
```
Comment (evshi) [inline packages/runtime/src/widgets/DataGrid/DataGrid.tsx:260]: nit: rename to better variables.

**Review by evshi — APPROVED** [review]

**Pull request merged**